drift-transport-netty/src/main/java/com/facebook/drift/transport/netty/codec/HeaderTransport.java

```diff
@@ -200,6 +235,94 @@ public static ThriftFrame decodeFrame(ByteBuf buffer)
         }
     }
 
+    private static ByteBuf transform(ByteBuf data, List<Transform> writeTransforms)
+    {
+        if (writeTransforms.contains(ZLIB_TRANSFORM)) {
```

Comment:
The spec does not say that actually but I would assume so. I didn't put much effort in this part since we only need to handle one simple transform at the moment. But it is definitely better make it more structured. The difficulty is some transform (though not ZLIB) will not only affect the payload but also header, so it is better to design around it when we need to.